### PR TITLE
Increased code coverage of CachePartitionLostListenerConfig

### PR DESCRIPTION
**Pull request by Donnerbart**
* Increased code coverage of `CachePartitionLostListenerConfig`.
* Cleanup of `CachePartitionLostListenerConfigTest`.

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/config/CachePartitionLostListenerConfig.java [hunk 1]
```diff
@@ -74,12 +74,7 @@ public boolean equals(Object o) {
         }
 
         CachePartitionLostListenerConfig that = (CachePartitionLostListenerConfig) o;
-
-        if (className != null ? !className.equals(that.className) : that.className != null) {
```
Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/config/CachePartitionLostListenerConfig.java]: This code was redundant as it's already checked in the `super.equals()` method. I added tests for this, so the removal should be fine. This dead code just hurts the code coverage.

**Comment by Donnerbart** [comment]
run-lab-run

**Comment by Donnerbart** [comment]
verify

**Comment by Donnerbart** [comment]
verify

**Comment by Donnerbart** [comment]
verify

**Comment by kobalski** [comment]
:+1:

**Comment by mesutcelik** [comment]
:+1:

**Pull request merged**